### ISSUE TITLE
Content block deleted

### ISSUE THREAD

**Issue by NeleDeBruycker** [spicywebau/craft-neo#939]
### Bug Description

With the latest update (5.2.4), all the content block in a Neo matrix seems to be deleted, causing an empty website. 

### Steps to reproduce

- Update Neo plugin to 5.2.4

### Neo version

5.2.4

### Craft CMS version

5.4.1

### Does this issue involve templating, and if so, is eager-loading used?

This is not a templating issue

**Comment by ishetnogferre** [comment]
Colleague of @NeleDeBruycker here: we did some log searching and found out, **all** our neo blocks got cleared because of the garbage collection that run after this was added: https://github.com/spicywebau/craft-neo/commit/5c78978ffaedd9ce15ef249f16daf26c3b18e664

**Comment by ttempleton** [comment]
Not quite sure why that has happened; safe to say if I could reproduce it, I wouldn't have made the release.

If possible, please send your Composer files and database backup from before garbage collection to plugins@spicyweb.com.au along with information about an entry this is happening to, and we'll have a look.

**Comment by ishetnogferre** [comment]
Unfortunately the db is quite big so can't really send it over. Just tested it locally with the same code in a module:

```
$elementsOwnersRows = (new Query())
                ->select([
                    'eo.elementId',
                    'eo.ownerId',
                    'eo.sortOrder',
                ])
                ->from(['eo' => Table::ELEMENTS_OWNERS])
                ->innerJoin(['nb' => '{{%neoblocks}}'], '[[nb.id]] = [[eo.elementId]]')
                ->innerJoin(['nbs' => '{{%neoblockstructures}}'], '[[nbs.ownerId]] = [[eo.ownerId]]')
                ->leftJoin(['se' => Table::STRUCTUREELEMENTS], [
                    'and',
                    '[[nbs.structureId]] = [[se.structureId]]',
                    '[[eo.elementId]] = [[se.elementId]]',
                ])
                ->where(['se.structureId' => null])
                ->limit(500)
                ->all();

        Craft::dd($elementsOwnersRows);
```

Without the limit I added, the page crashes, if I add a limit, that's the amount of items I get returned

![Screenshot 2024-09-06 at 11 58 05](https://github.com/user-attachments/assets/7702374d-0b4a-481a-b288-d62295b04f28)

I'm gonna check and see if I can recreate it in another project so we can send that one over.

**Comment by ttempleton** [comment]
Actually, is it a multi-site Craft install? I'm away from my dev machine at the moment so can't test right now, but wondering if that could be related.

**Comment by ishetnogferre** [comment]
@ttempleton it is, we also tested it on another (multisite) Craft 5 with neo installed and there it also returns I think almost all blocks, so I guess there might be something in the query that isn't correct yet?

**Comment by ttempleton** [comment]
Just released 5.2.5, which now checks if any structure element data exists for a given block/owner, instead of checking for a null structure ID (which would return results we don't want to delete on multi-site Craft installs if the propagation method was set to something other than to save blocks to all sites).

Sorry about this @NeleDeBruycker @ishetnogferre and please let me know if any issues persist.

**Comment by ishetnogferre** [comment]
@ttempleton this seems to have fixed it, thanks for the quick response and fix! 🙌
